src/crepe/api/AI.cpp

```diff
@@ -0,0 +1,87 @@
+#include <stdexcept>
+
+#include "AI.h"
+
+namespace crepe {
+
+AI::AI(game_object_id_t id, float max_force) : Component(id), max_force(max_force) {}
+
+void AI::make_circle_path(float radius, const vec2 & center, float start_angle,
+						  bool clockwise) {
+	if (radius <= 0) {
+		throw std::runtime_error("Radius must be greater than 0");
+	}
+
+	// The step size is determined by the radius (step size is in radians)
+	float step = RADIUS_TO_STEP / radius;
+	// Force at least MIN_STEP steps (in case of a small radius)
+	if (step > 2 * M_PI / MIN_STEP) {
+		step = 2 * M_PI / MIN_STEP;
+	}
+	// The path node distance is determined by the step size and the radius
+	path_node_distance = radius * step * 0.75f;
+
+	if (clockwise) {
+		for (float i = start_angle; i < 2 * M_PI + start_angle; i += step) {
+			path.push_back(vec2{static_cast<float>(center.x + radius * cos(i)),
+								static_cast<float>(center.y + radius * sin(i))});
+		}
+	} else {
+		for (float i = start_angle; i > start_angle - 2 * M_PI; i -= step) {
+			path.push_back(vec2{static_cast<float>(center.x + radius * cos(i)),
+								static_cast<float>(center.y + radius * sin(i))});
+		}
+	}
+}
+
+void AI::make_oval_path(float radius_x, float radius_y, const vec2 & center, float start_angle,
+						bool clockwise, float rotation) {
+	if (radius_x <= 0 && radius_y <= 0) {
+		throw std::runtime_error("Radius must be greater than 0");
+	}
+
+	float max_radius = std::max(radius_x, radius_y);
+	// The step size is determined by the radius (step size is in radians)
+	float step = RADIUS_TO_STEP / max_radius;
+	// Force at least MIN_STEP steps (in case of a small radius)
+	if (step > 2 * M_PI / MIN_STEP) {
+		step = 2 * M_PI / MIN_STEP;
+	}
+	// The path node distance is determined by the step size and the radius
+	path_node_distance = max_radius * step * 0.75f;
+
+	auto rotate_point = [rotation](vec2 point, vec2 center) {
```

Comment:
Is dit een specfieke vec2 functie?
Als dit zo is moet die in de vec2 klasse.

Maar het lijkt er meer op dat je dit gebruikt om een punt uit te rekenen dus specifiek voor AI.
En je hebt het een lambda gemaakt zodat je de interface niet aanpast van AI en nog steeds aan de ecs component voldoet.
Als het bovenstaande geval is wil je dan de auto weg halen?